pyanaconda/dbus/template.py

```diff
@@ -0,0 +1,99 @@
+#
+# template.py: Templates for DBus objects.
+#
+# Copyright (C) 2017  Red Hat, Inc.  All rights reserved.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+from pyanaconda.dbus import DBus
+
+__all__ = ["InterfaceTemplate"]
+
+
+class InterfaceTemplate(object):
```

Comment:
What about using ABC here and made the `connect_signals` method an abstract method.

Reply:
Used ABC.